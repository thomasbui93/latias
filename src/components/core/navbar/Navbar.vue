<template>
  <div class="navbar has-shadow is-spaced" role="navigation" aria-label="main navigation">
    <div class="navbar-brand">
      <div class="navbar-item">
        Latias
      </div>
      <a role="button" class="navbar-burger burger" aria-label="menu" aria-expanded="false">
        <span aria-hidden="true"></span>
        <span aria-hidden="true"></span>
        <span aria-hidden="true"></span>
      </a>
    </div>
    <div class="navbar-menu">
      <div class="navbar-end is-hidden-mobile">
        <div class="navbar-item">
          <div class="buttons">
            <router-link to="/login">
              <a class="button is-primary">
                <strong>Sign in</strong>
              </a>
            </router-link>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>
