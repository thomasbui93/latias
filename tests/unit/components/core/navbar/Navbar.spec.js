import { shallowMount, createLocalVue } from '@vue/test-utils';
import Navbar from '@/components/core/navbar/Navbar.vue';
import VueRouter from 'vue-router';

const localVue = createLocalVue();
localVue.use(VueRouter);

describe('Navbar component', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(Navbar, {
      stubs: ['router-link', 'router-view'],
      propsData: {
        isAuthenticated: true,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render sign in button if isAuthenticated is false', () => {
    const wrapper = shallowMount(Navbar, {
      stubs: ['router-link', 'router-view'],
      propsData: {
        isAuthenticated: false,
      },
    });
    expect(wrapper.findAll('.login').length).toBe(1);
    expect(wrapper.findAll('.logout').length).toBe(0);
  });
});
