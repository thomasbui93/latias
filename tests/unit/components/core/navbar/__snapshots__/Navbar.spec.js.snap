// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component matches snapshot 1`] = `
<div role="navigation" aria-label="main navigation" class="navbar has-shadow is-spaced" isauthenticated="true">
  <div class="navbar-brand">
    <div class="navbar-item">
      <router-link-stub to="/" class="logo">Latias</router-link-stub>
    </div> <a role="button" aria-label="menu" aria-expanded="false" class="navbar-burger burger"><span aria-hidden="true"></span> <span aria-hidden="true"></span> <span aria-hidden="true"></span></a>
  </div>
  <div class="navbar-menu">
    <div class="navbar-end is-hidden-mobile">
      <div class="navbar-item">
        <div class="buttons">
          <router-link-stub to="/sign-in"><a class="button is-primary login"><strong>Sign in</strong></a></router-link-stub>
          <!---->
        </div>
      </div>
    </div>
  </div>
</div>
`;
